SDK/bundles/com.ashling.riscfree.debug.opxd.registers/src/com/ashling/riscfree/debug/opxd/registers/core/RiscFreeRegister.java

```diff
@@ -0,0 +1,749 @@
+		return config;
+	}
+
+	public String getRegisterFile() {
+
+		String tempFileName = null;
+		try {
+			SAXParserFactory spf = SAXParserFactory.newInstance();
+			spf.setXIncludeAware(true);
+			spf.setNamespaceAware(true);
+			XMLReader xr = spf.newSAXParser().getXMLReader();
+			// be sure validation is "off" or the feature to ignore DTD's will not apply
+			xr.setFeature("http://xml.org/sax/features/validation", true); //$NON-NLS-1$
+			xr.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", true); //$NON-NLS-1$
+			xr.setEntityResolver((p, s) -> {
+				if (s.contains("gdb-target.dtd")) {
+					return new InputSource(new FileInputStream(dtdLocation + "\\gdb-target.dtd"));
+				}
+				if (s.contains("xinclude.dtd")) {
+					return new InputSource(new FileInputStream(dtdLocation + "\\xinclude.dtd"));
+				}
+				return null;
+			});
+			FileInputStream xmlStream = new FileInputStream(rootRegisterFile);
+			SAXSource src = new SAXSource(xr, new InputSource(xmlStream));
+			src.setSystemId(registerDirectory);
+			JAXBContext jaxbContext = JAXBContext.newInstance(Target.class);
+			Unmarshaller unMarshaller = jaxbContext.createUnmarshaller();
+
+			Target target = (Target) unMarshaller.unmarshal(src);
+			// Target target = JAXB.unmarshal(src, Target.class);
+			Map<String, Object> typeMap = new HashMap<>();
+			target.getFeature().forEach(feature -> {
+				feature.getVectorOrFlagsOrStructOrUnionOrEnum().forEach(type -> {
+					if (type instanceof Vector) {
+						typeMap.put(((Vector) type).getId(), type);
+					} else if (type instanceof Flags) {
+						typeMap.put(((Flags) type).getId(), type);
+					} else if (type instanceof Struct) {
+						typeMap.put(((Struct) type).getId(), type);
+					} else if (type instanceof Union) {
+						typeMap.put(((Union) type).getId(), type);
+					} else if (type instanceof Enum) {
+						typeMap.put(((Enum) type).getId(), type);
+					}
+				});
+			});
+			target.getFeature().forEach(feature -> {
+				feature.getReg().forEach(reg -> {
+					if (typeMap.containsKey(reg.getType())) {
+						if (reg.getBitsize().equals("32")) {
+							reg.setType("uint32");
+						} else if (reg.getBitsize().equals("64")) {
+							reg.setType("uint64");
+						}
+					}
+				});
+			});
+			Marshaller marshaller = jaxbContext.createMarshaller();
+			marshaller.setProperty(Marshaller.JAXB_FRAGMENT, Boolean.TRUE);
+			java.io.StringWriter sw = new StringWriter();
+			marshaller.marshal(target, sw);
+
+			String str = fileToString(rootRegisterFile);
+			//To remove commented code from xml
+			str=str.replaceAll( "(?s)<!--.*?-->", "" );
+			String replaceString = str.substring(str.indexOf("<target"), str.indexOf("</target>") + 9);
+			tempFileName = dtdLocation +File.separator+ new SimpleDateFormat("yyyyMMddHHmm'.xml'").format(new Date());
+			StringToFile(tempFileName, str.replace(replaceString, sw.toString()));
+
+		} catch (Exception e) {
+			e.printStackTrace();
```

Comment:
If we suppress the exception here, what would happen? this doesn't look good.

Reply:
Changes are done.